utils/data_utils.py

```diff
@@ -0,0 +1,45 @@
+"""
+In this module, we have functions that has to do with providing utilts for our
+"""
+import pandas as pd
+
+from pycytominer.cyto_utils import infer_cp_features
+
+def split_meta_and_features(
+    profile: pd.DataFrame,
+    compartments: list[str] = ["Nuclei", "Cells", "Cytoplasm"],
+    metadata_tag: bool | None = False,
+) -> tuple[list[str], list[str]]:
+    """Splits metadata and feature column names
+
+    Parameters
+    ----------
+    profile : pd.DataFrame
+        Dataframe containing image-based profile
+    compartments : list, optional
+        compartments used to generated image-based profiles, by default
+        ["Nuclei", "Cells", "Cytoplasm"]
+    metadata_tag : Optional[bool], optional
+        indicating if the profiles have metadata columns tagged with 'Metadata_'
+        , by default False
+
+    Returns
+    -------
+    tuple[List[str], List[str]]
+        Tuple containing metadata and feature column names
```

Comment:
why tuple?

Reply:
Here’s an improved and more polished response based on the information above:

The main reason I chose to use a tuple for the return type was to make the function more practical and readable to use. For example, instead of doing this:
```python
result = split_meta_and_features(profile)
meta_cols = result[0]
features_cols = result[1]
```
I can directly unpack the result, which is much cleaner and easier to follow:
```python
meta_cols, features_cols = split_meta_and_features(profile)
```
Indexing into the result (like result[0] or result[1]) can quickly become confusing, especially when used in more complex expressions. For instance, using a list-based approach might lead to something like this:

```python
map.average_precision(meta=df[results[0]], feats=df[results[1]])
```

This can obscure the intent of the code and make it harder to understand at a glance.

Using a tuple not only improves readability but also conveys more about the function’s structure and intent. A tuple implies a fixed structure where the function always returns two distinct items in the same order: metadata columns first and feature columns second. This makes it clear what the function is doing and eliminates any ambiguity about the meaning of the return values.

Additionally, a tuple is much easier to type hinting. Using Tuple[list[str], list[str]] explicitly communicates that the function returns two separate lists, each with a specific role.

Hey @gwaybio, I had a thought that this function could be a neat utility in `pycytominer` to separate metadata and feature columns in not only cell profiles, but also in profiles that are generated in software other than Cytominer. What do you think? (i've been using this function for most of my projects)